kurtosis-devnet/pkg/kurtosis/kurtosis.go

```diff
@@ -0,0 +1,379 @@
+
+	endpointMap := make(EndpointMap)
+	var nodes []Node
+
+	for serviceName, ports := range services {
+		var port int
+		for _, interestingPort := range interestingPorts {
+			if p, ok := ports[interestingPort]; ok {
+				port = p
+				break
+			}
+		}
+		if port == 0 { // nothing to see here
+			continue
+		}
+
+		if serviceIdentifier, num, ok := matchService(serviceName); ok {
+			var allocated bool
+			for _, service := range nodeServices {
+				if serviceIdentifier == service { // this is a node
+					if num > len(nodes) {
+						nodes = append(nodes, make(Node))
+					}
+					nodes[num-1][serviceIdentifier] = fmt.Sprintf("http://localhost:%d", port)
+					allocated = true
+				}
+			}
+			if !allocated {
+				endpointMap[serviceIdentifier] = fmt.Sprintf("http://localhost:%d", port)
+			}
+		}
+	}
+	return nodes, endpointMap
+}
+
+// return the shorthand service tag (used as key in the final output) and the
+// index if that's a service with multiple instances.
+func serviceTag(serviceName string) (string, int) {
+	// Find index of first number
+	i := strings.IndexFunc(serviceName, func(r rune) bool {
+		return r >= '0' && r <= '9'
+	})
+	if i == -1 {
+		return serviceName, 0
+	}
+	idx, err := strconv.Atoi(serviceName[i : i+1])
+	if err != nil {
+		return serviceName, 0
+	}
+	return serviceName[:i-1], idx
+}
+
+const l2ServiceTagPrefix = "op-"
+
+func findL2Endpoints(services inspect.ServiceMap, suffix string) ([]Node, EndpointMap) {
+	return findRPCEndpoints(services, func(serviceName string) (string, int, bool) {
+		if strings.HasSuffix(serviceName, suffix) {
+			name := strings.TrimSuffix(serviceName, suffix)
+			tag, idx := serviceTag(strings.TrimPrefix(name, l2ServiceTagPrefix))
+			return tag, idx, true
+		}
+		return "", 0, false
+	})
+}
+
+// TODO: L1 services are detected as "non-L2" right now. That might need to change
+// in the future, but for now it's good enough.
+func findL1Endpoints(services inspect.ServiceMap) ([]Node, EndpointMap) {
+	return findRPCEndpoints(services, func(serviceName string) (string, int, bool) {
+		match := !strings.HasPrefix(serviceName, l2ServiceTagPrefix)
+		if match {
+			tag, idx := serviceTag(serviceName)
+			return tag, idx, true
+		}
+		return "", 0, false
+	})
+}
+
+// prepareArgFile creates a temporary file with the input content and returns its path
+// The caller is responsible for deleting the file.
+func (d *KurtosisDeployer) prepareArgFile(input io.Reader) (string, error) {
+	argFile, err := os.CreateTemp("", "kurtosis-args-*.yaml")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary arg file: %w", err)
+	}
+	defer argFile.Close()
+
+	if _, err := io.Copy(argFile, input); err != nil {
+		os.Remove(argFile.Name())
+		return "", fmt.Errorf("failed to write arg file: %w", err)
+	}
+
+	return argFile.Name(), nil
+}
+
+// runKurtosisCommand executes the kurtosis command with the given arguments
+// TODO: reimplement this with the kurtosis SDK, it'll be cleaner.
```

Comment:
This is already implemented using the SDK - check out the kurtosis runner in op-e2e.

Reply:
ack